lib/csv/preprocessing.ts

```diff
@@ -0,0 +1,140 @@
+import { DataUploadEnum } from '@/lib/types'
+import { CsvError, parse } from 'csv-parse/sync'
+import DataFrame from 'dataframe-js'
+
+const uploadTypeProcessFunctionMap = {
+  [DataUploadEnum.APPLICANT]: processApplicantData,
+  [DataUploadEnum.TMUA_SCORES]: processTMUAData,
+  [DataUploadEnum.USER_ROLES]: processUserData
+}
+
+/**
+ * Parses a CSV and runs processing function according to the upload type
+ * @param file - the CSV file to process
+ * @param uploadType - the type of data being uploaded
+ * @returns flag to indicate success with data if true and errorMessage if false
+ */
+export async function preprocessCsvData(
+  file: File,
+  uploadType: DataUploadEnum
+): Promise<{ success: true; data: unknown[] } | { success: false; errorMessage: string }> {
+  if (file.name.split('.').pop() !== 'csv')
```

Comment:
Fine for now, but probably a little weak: could be worth checking via mime types instead

Reply:
MIME type is a better approach, will change

Reply:
Updated validation in e1b9ea86812472ac8254a641ee6496fc6e4cda70